### PR TITLE
Add tests for sentiment api

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
$subject

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: back-end/sentiment-api/resources/client_private.key [hunk 1]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (TharmiganK) [inline back-end/sentiment-api/resources/client_private.key]: These keys are used to mock the sts server right? Should we include them in the `tests/resources` directory?

File: back-end/sentiment-api/tests/auth_service.bal [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import ballerina/http;
```
Comment (TharmiganK) [inline back-end/sentiment-api/tests/auth_service.bal]: Can we rename this file as `mock_sts_endpoint`

File: back-end/sentiment-api/tests/sentiment_service_test.bal [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                }
+            }
+        }
+    );
+}
+
+@test:Config {}
+function testSentimentApiWhenUnavailable() returns error? {
+    Sentiment|http:Error sentiment = (<http:Client>sentimentClient)->/api/sentiment.post({text: "I am happy!"});
+    test:assertTrue(sentiment is http:Error, "sentiment response is not an error");
```
Comment (TharmiganK) [inline back-end/sentiment-api/tests/sentiment_service_test.bal]: Isn't this more readable?
```bal
    if sentiment is http:Error {
        test:assertEquals(sentiment.message(), "Service Unavailable");
    }
    test:assertFail("sentiment response is not an error");
```

Comment (gayaldassanayake) [reply back-end/sentiment-api/tests/sentiment_service_test.bal]: Changed to,
``` ballerina
if sentiment is Sentiment {
        test:assertFail("sentiment response is not an error");
}
test:assertEquals(sentiment.message(), "Service Unavailable");
```

with c33cde2bebadeb76270aa8fa8add15e6a8ea6a62

File: back-end/sentiment-api/tests/auth_service.bal (hunk 2 above)
Comment (gayaldassanayake) [reply back-end/sentiment-api/tests/auth_service.bal]: Addressed with c33cde2bebadeb76270aa8fa8add15e6a8ea6a62

File: back-end/sentiment-api/resources/client_private.key (hunk 1 above)
Comment (gayaldassanayake) [reply back-end/sentiment-api/resources/client_private.key]: Addressed with c33cde2bebadeb76270aa8fa8add15e6a8ea6a62

**Review by TharmiganK — APPROVED** [review]
LGTM

**Review by azinneera — APPROVED** [review]

**Pull request merged**